src/core/cpu_patches.cpp

```diff
@@ -108,9 +118,7 @@ static Xbyak::Reg AllocateScratchRegister(
     UNREACHABLE_MSG("Out of scratch registers!");
 }
 
-#ifdef __APPLE__
```

Comment:
This was very intentionally ifdef'd out on non-Apple as it relies on their conventions for TLS in the GS segment register.

Reply:
Hmm in that case I could push/pop instead of using the Allocate functions

Reply:
From my experiements writing this system you cannot reliably use the stack or any registers without saving them somewhere first. The stack because some games have functions that use stack area below the stack pointer, so you don't have any way to know for sure whether you're clobbering data or not.

My solution was to save registers to TLS (and I have some code that saves the stack to TLS and allows arbitrary use of the stack to save registers), but it still relies on the Apple TLS convention as you can do a simple write to gs + offset without any scratch registers.

Reply:
Just to be clear, if there's a way to implement the save on other OSes without any scratch space that would resolve the issue. The rest of the code shouldn't have any OS-specific issues.